lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -224,12 +235,12 @@ if Enum.all?(
       {_removed_track, tracks} = Map.pop!(state.tracks, track_id)
       state = %{state | tracks: tracks}
 
-      children_to_remove = add_video_sink_children_if_no_video_left(children_to_remove, tracks)
+      children_to_remove = get_children_to_remove_if_no_track_left(children_to_remove, tracks)
 
       {{:ok, remove_child: children_to_remove}, state}
     end
 
-    defp add_video_sink_children_if_no_video_left(children, []),
+    defp get_children_to_remove_if_no_track_left(children, tracks) when map_size(tracks) == 0,
       do:
         children ++
           [
```

Comment:
It has to include `aac_encoder` and `aac_parser` because we pass to `audio_mixer` raw audio and then this audio must be encoded and parsed. So in the end `aac_encoder` and `aac_parser` must be removed at the same time as `audio_mixer`.